pacwrapper.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+	"text/template"
+)
+
+type PACData struct {
+	Port     int
+	PACURL   string
+	Domain   string
+	Username string
+}
+
+type pacData struct {
+	PACData
+	PAC string
```

Comment:
The name "PAC" feels a little ambiguous to me, can we make it something more descriptive? How about UpstreamPAC?

Reply:
Done. Fixed in eccb52048c68062948af4e90bdc3b6d295c8d47b